handler/src/main/java/io/netty/handler/ssl/OpenSslCertificateException.java

```diff
@@ -70,11 +70,80 @@ public int errorCode() {
     }
 
     private static int checkErrorCode(int errorCode) {
-        if (errorCode < CertificateVerifier.X509_V_OK || errorCode > CertificateVerifier.X509_V_ERR_DANE_NO_MATCH) {
-            throw new IllegalArgumentException("errorCode must be " + CertificateVerifier.X509_V_OK + " => "
-                    + CertificateVerifier.X509_V_ERR_DANE_NO_MATCH +
-                    ". See https://www.openssl.org/docs/manmaster/apps/verify.html .");
+        if (!isValid(errorCode)) {
+            throw new IllegalArgumentException("errorCode '" + errorCode +
+                    "' invalid. See https://www.openssl.org/docs/manmaster/apps/verify.html .");
         }
         return errorCode;
     }
+
+    // Package-private for testing
+    static boolean isValid(int errorCode) {
+        return errorCode == CertificateVerifier.X509_V_OK ||
+                errorCode == CertificateVerifier.X509_V_ERR_UNSPECIFIED ||
```

Comment:
@Scottmitch I am not sure if this should life here or if we should better move it to tcnative. WDYT ?